### ISSUE TITLE
FedProx: Docstring likely missing .detach() on how to adjust the client-side loss function with proximal term

### ISSUE THREAD

**Issue by laminair** [adap/flower#1774]
### Describe what the documentation is missing.

I recently added the FedProx strategy to my project and made adjustments to my PyTorch models exactly as provided in the docstring of the FedProx strategy: https://flower.dev/docs/_modules/flwr/server/strategy/fedprox.html
Yet, I received a `requires_grad` error originating from `local_weights` not being detached. 

### Suggest your improvement.

Detaching local_weight tensors solved the issue for me.
Maybe it's worth adjusting in the docstring like so: 

```
.. code:: python

          for local_weights, global_weights in zip(net.parameters(), global_params):
              proximal_term += (local_weights.detach() - global_weights).norm(2)    # <--- Here I added the .detach()
          loss = criterion(net(inputs), labels) + (config["proximal_mu"] / 2) * proximal_term
```

**Comment by WilliamLindskog** [comment]
Hi

Thanks for raising this. This seems to have been resolved in newer flwr versions. I tested using the `fedProx` strategy and the `requires_grad` error is not thrown. 

Closing this for now.
